test/pages/AddBookPage.ts

```diff
@@ -0,0 +1,39 @@
+import { Locator, Page } from "@playwright/test";
+
+export class AddBookPage {
+  page: Page;
+  profileIcon: Locator;
+  addBookLink: Locator;
+  titleLocator: Locator;
+  publisherLocator: Locator;
+  publishedLocator: Locator;
+  addButton: Locator;
+
+  constructor({ page }: { page: Page }) {
+    this.page = page;
+    this.titleLocator = page.getByRole("textbox", { name: "Title" });
+    this.publisherLocator = page.getByRole("textbox", {
+      name: "Who is the publisher",
+    });
+    this.publishedLocator = page.getByRole("textbox", {
+      name: "When was it published?",
+    });
+    this.addButton = page.getByRole("button", { name: "Add" });
+    this.profileIcon = page.getByAltText("My account");
+    this.addBookLink = page.locator(
+      `a[data-ol-link-track="Hamburger|AddBook"]`
+    );
+  }
+
+  async navigateToAddBookPage() {
```

Comment:
the page is already called `AddBookPage`, so no need to repeat it in the function name
```suggestion
  async navigate() {
```